addons/html_editor/static/src/main/link/link_plugin.js

```diff
@@ -242,11 +252,60 @@ export class LinkPlugin extends Plugin {
     }
 
     /**
-     * Remove the link from the selection
+     * Remove the link from the collapsed selection
      */
     removeLink() {
         const link = this.linkElement;
-        unwrapContents(link);
+        const cursors = this.shared.preserveSelection();
+        if (link && link.isContentEditable) {
+            unwrapContents(link);
```

Comment:
consider including `cursors.update(callbacksForCursorUpdate.unwrap(p));` before this line (once https://github.com/odoo-dev/odoo/pull/3149/files is merged).